integration/network.go

```diff
@@ -0,0 +1,47 @@
+package integration
+
+import (
+	"fmt"
+	"net"
+)
+
+const (
+	portRangeMin = 0
+	portRangeMax = 65535
+)
+
+var (
+	errPortMax = portError{
+		message: fmt.Sprintf("port cannot be greater than %d", portRangeMax),
+	}
+	errPortMin = portError{
+		message: fmt.Sprintf("port cannot be less than %d", portRangeMin),
+	}
+	errPortNotFound = portError{
+		message: fmt.Sprintf("no open port found %d", portRangeMin),
+	}
+)
+
+type portError struct{ message string }
+
+// Error returns a string with the port-related error message
+func (p portError) Error() string {
+	return fmt.Sprintf("port discovery error: %s", p.message)
+}
+
+// GetOpenPortInRange finds the first unused port within specific range
+func GetOpenPortInRange(lowerBound, upperBound int) (int, error) {
```

Comment:
it'd be great to have an even simpler function that takes no parameters and returns an int -- when I'm using this I usually get confused about what port range I should care about and then end up copying/pasting something from somewhere else. A default range would be nice to have

Reply:
Maybe something like this?

```
// GetOpenPort finds an unused port
func GetOpenPort() {
	GetOpenPortInRange(portRangeMin, portRangeMax)
}

// GetOpenPortInRange finds the first unused port within specific range
func GetOpenPortInRange(lowerBound, upperBound int) (int, error) {
...
}
```

Reply:
I like the API, but don't like ignoring the error. Maybe either take in a `testing.T` or some `Fataller` interface that will let you call `.Fatal(interface{})`?

Reply:
> I like the API, but don't like ignoring the error.

Agreed. I left that out by accident.

> Maybe either take in a testing.T or some Fataller interface that will let you call .Fatal(interface{})?

Yeah, I was wondering about the whole `fatal` notion. I initially passed a `testing.T` to a number of these helpers, but removed the `testing.T` argument because IMO these helpers should not make the decision to fatal. The alternative is to call these helpers inside a `_test.go` file and have developers choose how they want to handle errors.

Whatever we decide, I think we should go "all or nothing" in favor of that decision. As in, all helpers should take a `testing.T` or none of them should. So, with that in mind, we should consider whether or not it makes sense for the other helpers to take a `testing.T` type.